crates/common/src/rpc/codec.rs

```diff
@@ -0,0 +1,20 @@
+// rmp_serde and bincode have various limitations and are troublesome to debug.
+// the overhead of JSON for messages is not a concern for the RPC layer.
+
+pub fn encode<T: serde::Serialize>(msg: &T) -> serde_json::Result<Vec<u8>> {
+    serde_json::to_vec(msg)
+}
+
+pub fn decode<'de, T: serde::Deserialize<'de>>(msg: &'de [u8]) -> serde_json::Result<T> {
+    serde_json::from_slice(msg)
+}
+
```

Comment:
Do we still need these if we are removing `bincode`? Are we also full swapping from `tarpc` to a json one? Or will that be another PR?

Reply:
We can swap to a real jsonrpc in another PR